crates/bevy_sprite/src/render/mod.rs

```diff
@@ -533,210 +622,140 @@ pub fn queue_sprites(
-                                            ),
-                                        },
-                                        BindGroupEntry {
-                                            binding: 1,
-                                            resource: BindingResource::Sampler(&gpu_image.sampler),
-                                        },
-                                    ],
-                                    label: Some("sprite_material_bind_group"),
-                                    layout: &sprite_pipeline.material_layout,
-                                })
-                            });
-                    } else {
-                        // Skip this item if the texture is not ready
-                        continue;
-                    }
-                }
-
-                // Calculate vertex data for this item
-
-                let mut uvs = QUAD_UVS;
-                if extracted_sprite.flip_x {
-                    uvs = [uvs[1], uvs[0], uvs[3], uvs[2]];
-                }
-                if extracted_sprite.flip_y {
-                    uvs = [uvs[3], uvs[2], uvs[1], uvs[0]];
-                }
+            for item_index in 0..transparent_phase.items.len() {
+                let item = &transparent_phase.items[item_index];
+                if let Ok(extracted_sprite) = extracted_sprites.get(item.entity) {
+                    // Take a reference to an existing compatible batch if one exists
+                    let existing_batch = batches.last_mut().filter(|_| {
+                        batch_image_handle == extracted_sprite.image_handle_id
+                            && batch_colored == (extracted_sprite.color != Color::WHITE)
+                    });
 
-                // By default, the size of the quad is the size of the texture
-                let mut quad_size = current_image_size;
+                    // Either keep the reference to the compatible batch or create a new batch
+                    let (_, target_batch) = match existing_batch {
+                        Some(batch) => batch,
+                        None => {
+                            // We ensure the associated image exists in queue_sprites
+                            let gpu_image = gpu_images
+                                .get(&Handle::weak(extracted_sprite.image_handle_id))
+                                .unwrap();
+
+                            batch_item_index = item_index;
+                            batch_image_size = Vec2::new(gpu_image.size.x, gpu_image.size.y);
+                            batch_image_handle = extracted_sprite.image_handle_id;
+                            batch_colored = extracted_sprite.color != Color::WHITE;
+
+                            let new_batch = SpriteBatch {
+                                range: if batch_colored {
+                                    colored_index..colored_index
+                                } else {
+                                    index..index
```

Comment:
The end for each of these is updated after the vertices are added to the buffers so the end value isn't important. This change would make the type RangeInclusive instead of Range so not doing this immediately but if there's a reason I'm not seeing am open to suggestions.

Reply:
Nope. Got it.